frontend/src/components/userdashboard/quizPage.js

```diff
@@ -1,15 +1,54 @@
 import { useState, useEffect } from "react";
+import { useParams } from "react-router-dom";
+import axios from "../api/api";
 
+const CAT_URL = '/categories/';
+const LESS_URL = '/lessons/';
 const Quizitem = () => {
 
-    const [ans, setAns] = useState(false);
+    const categoryId = useParams().categoryId;
+    const [ans, setAns] = useState([]);
+    const [title, setTitle] = useState('');
+    const [word, setWord] = useState([]);;
+    const [counter, setCounter] = useState(0);
 
     useEffect(() => {
         setAns('');
     }, [])
 
+    useEffect(() => {
+        fetchTitle(categoryId);
+    }, [])
+
+    useEffect(() => {
+        fetchWord(categoryId);
+    }, [])
```

Comment:
You can insert all functions inside of one useEffect if the dependencies are none for all.
If you're going to depend on some state please include it via comment.

```
 useEffect(() => {
        setAns('');
        fetchTitle(categoryId);
        fetchWord(categoryId);
    }, []) 
```